tests/load-tests/run-max-concurrency.sh

```diff
@@ -73,14 +73,20 @@ else
         index=$(printf "%04d" "$t")
         cp -vf "$output_dir/load-tests.json" "$output_dir/load-tests.max-concurrency.$index.json"
         cp -vf "$output_dir/load-tests.log" "$output_dir/load-tests.max-concurrency.$index.log"
-        pipelineRunThresholdExceeded=$(jq -rc ".runPipelineSucceededTimeMax > $threshold" "$output_dir/load-tests.json")
-        pipelineRunKPI=$(jq -rc ".runPipelineSucceededTimeMax" "$output_dir/load-tests.json")
-        if [ "$pipelineRunThresholdExceeded" = "true" ]; then
-            echo "The maximal time a pipeline run took to succeed (${pipelineRunKPI}s) has exceeded a threshold of ${threshold}s with $t threads."
+        workloadKPI=$(jq '.workloadKPI' "$output_dir/load-tests.json")
+        if awk "BEGIN { exit !($workloadKPI > $threshold)}"; then
+            echo "The average time a workload took to succeed (${workloadKPI}s) has exceeded a threshold of ${threshold}s with $t threads."
+            indexOld=$(printf "%04d" "$(($t - 1))")
+            workloadKPIOld=$(jq '.workloadKPI' "$output_dir/load-tests.max-concurrency.$indexOld.json")
+            threadsOld=$(jq '.maxConcurrencyReached' "$output_dir/load-tests.max-concurrency.$indexOld.json")
+            computedConcurrency=$(python -c "import sys; t = float(sys.argv[1]); a = float(sys.argv[2]); b = float(sys.argv[3]); c = float(sys.argv[4]); d = float(sys.argv[5]); print((t - b) / ((d - b) / (c - a)) + a)" "$threshold" "$threadsOld" "$workloadKPIOld" "$t" "$workloadKPI")
+            jq ".computedConcurrency = $computedConcurrency" "$output" >"$output_dir/$$.json" && mv -f "$output_dir/$$.json" "$output"
```

Comment:
@jhutar the `.maxConcurrencyReached` in the final JSON (after the test has finished) is the number of threads before the threshold was crossed. It is the last number of threads from the `maxConcurrencySteps` list at which the `workloadKPI` was under the `threshold`.

It is not the value captured at the very last iteration where the threshold was exceeded.

Reply:
If you want to capture the values when it crossed the line, you should store it like:
```
.concurrencyAboveThreshold = $t
.workloadKPIAboveThreshold = $workloadKPI
```
in the section when the condition for exceeding the threshold is met

Reply:
![chart](https://github.com/redhat-appstudio/e2e-tests/assets/3397250/e0e8e3ac-9dca-4c3f-9880-63ee7b690f91)

Look here. This is real world example where we crossed 300 between runs for 130 and 140 threads (i.e. concurrent users). If I assume the dependency is linear (I think graph for all threads values confirms it), `computedConcurrency` now contains 136.something. Not ideal, but should be closer to truth than just 130 we used so far.

Reply:
Oh I see, make sense - I like the idea